### PR TITLE
Overseas rules for French housing benefits

### PR DESCRIPTION
**Pull request by denismerigoux**
Started with @JusBanuls!

### CODE REVIEW

File: examples/aides_logement/code_construction_legislatif.catala_fr [hunk 1]
```diff
@@ -1120,7 +1120,17 @@ soins de longue durée mentionnés au 3° de l'article L. 162-22 du code de la s
...
 champ d'application ÉligibilitéAllocationLogement:
-  exception définition éligibilité état l841_2 sous condition
+  exception accession_propriété
+  # Nous considérons que L841-3 est une exception à L841-4 et non pas à
+  # L841-2 (cas de base) car si L841-3 et L84164 étaient tous deux exceptions
+  # de L841-2, alors il pourrait y avoir conflit entre ces deux exceptions.
+  # Cependant le conflit est impossible puisqu'on ne peut pas à la fois
+  # être dans une unité de soin longue durée et avoir acheté cette unité
+  # de soin longue durée avec un prêt aidé par l'État (quoique...
+  # https://www.mediapart.fr/journal/france/160123/les-folles-economies-de-la-famille-ciotti-au-detriment-de-la-securite-sociale).
```
Comment (AltGr) [inline examples/aides_logement/code_construction_legislatif.catala_fr:1130]: :exploding_head:

**Comment by denismerigoux** [comment]
After a few months of efforts, we finally finished implementing the special rules for overseas housing benefits computation. Some legal questions are still pending answers from the relevant authorities but we can merge the bulk of the code now. More tests are also missing.

This PR also come with a bunch of improvements and fixes that ought to be reviewed. Considered the Catala code to have been reviewed, all the rest should be inspected carefully.

**Review by AltGr — APPROVED** [review]
The compiler changes seem ok to me; I haven't reviewed all the Catala code! :eyes: 
Do we need to add the .pdf files ?

**Comment by denismerigoux** [comment]
The PDF files contain useful legal information about housing benefits overseas so yes, it's a form of documentation.

**Pull request merged**